x-pack/plugins/ml/common/util/chart_utils.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CHART_TYPE, ChartType } from '../constants/charts';
+import type { SeriesConfigWithMetadata } from '../types/results';
+
+/**
+ * TODO check if these flags are still needed
+ */
+const EVENT_DISTRIBUTION_ENABLED = true;
+const POPULATION_DISTRIBUTION_ENABLED = true;
+
+/**
+ * Get the chart type based on its configuration
+ * @param config
+ */
+export function getChartType(config: SeriesConfigWithMetadata): ChartType {
+  let chartType: ChartType = CHART_TYPE.SINGLE_METRIC;
+
+  if (config.functionDescription === 'lat_long' || config.mapData !== undefined) {
+    return CHART_TYPE.GEO_MAP;
+  }
+
+  if (
+    EVENT_DISTRIBUTION_ENABLED &&
+    config.functionDescription === 'rare' &&
+    config.entityFields.some((f) => f.fieldType === 'over') === false
+  ) {
+    chartType = CHART_TYPE.EVENT_DISTRIBUTION;
+  } else if (
+    POPULATION_DISTRIBUTION_ENABLED &&
+    config.functionDescription !== 'rare' &&
+    config.entityFields.some((f) => f.fieldType === 'over') &&
+    config.metricFunction !== null // Event distribution chart relies on the ML function mapping to an ES aggregation
+  ) {
+    chartType = CHART_TYPE.POPULATION_DISTRIBUTION;
```

Comment:
I'm getting a `500` error with `Cannot read properties of undefined (reading 'length')"` when loading population charts.

Reply:
Thanks for testing, fixed in https://github.com/elastic/kibana/pull/128165/commits/32f847f0d490494d4833ff92c41a7831af669a93